noncer.go

```diff
@@ -0,0 +1,68 @@
+package oauth1
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+)
+
+// Noncer provides random nonce strings.
+type Noncer interface {
+	Nonce() string
+}
+
+// NoncerFunc is an adapter to allow the use of
+// ordinary functions as Noncers. If f is a function
+// with the appropriate signature, NoncerFunc(f) is a
+// Noncer that calls f.
+type NoncerFunc func() string
```

Comment:
Probably not needed, folks can write a struct. And most should never need to alter the noncer.

Reply:
Agreed.

Reply:
This is in there because it's how I originally implemented the Noncers (I didn't bother with separate types and just used NoncerFuncs).